crucible-mir-comp/src/Mir/Compositional/Clobber.hs

```diff
@@ -99,6 +99,30 @@ traverseFieldShape sym f shp0 rv0 = goField shp0 rv0
         rv'' <- f shp rv'
         return $ RV $ W4.justPartExpr sym rv''
 
+-- | Apply `f` to all of the `RegValue`s within a `VariantBranch`.
+--
+-- Note that this leaves the `Pred` of the `VariantBranch` unchanged, so using
+-- this to handle `EnumShape` can't change which variant of the enum is active.
+traverseVariantShape :: forall sym p t st fs tp0 rtp args ret.
+  (IsSymInterface sym, sym ~ W4.ExprBuilder t st fs, HasCallStack) =>
+  sym ->
+  (forall tp.
+    TypeShape tp ->
+    RegValue sym tp ->
+    OverrideSim (p sym) sym MIR rtp args ret (RegValue sym tp)) ->
+  VariantShape tp0 -> VariantBranch sym tp0 ->
+  OverrideSim (p sym) sym MIR rtp args ret (VariantBranch sym tp0)
+traverseVariantShape sym f (VariantShape flds) (VB pe) =
+  case pe of
+    W4.Unassigned -> return $ VB $ W4.Unassigned
```

Comment:
Redundant `$`:

```hs
   W4.Unassigned -> return $ VB W4.Unassigned
```

Reply:
Fixed in d5235253c9d8f0d9427d2666f4ac9dac028e12cd